Core/Core/Assignments/AssignmentList/ViewModel/AssignmentCellViewModel.swift

```diff
@@ -30,30 +30,52 @@ public class AssignmentCellViewModel: ObservableObject {
     }
 
     public var route: URL? { assignment.htmlURL }
+    var icon: UIImage { assignment.icon ?? .assignmentLine }
+    var name: String { assignment.name }
+    var submissionStatus: String { assignment.submission?.stateDisplayProperties.text ?? SubmissionStatus.notSubmitted.text }
+    var submissionIcon: UIImage { assignment.submission?.stateDisplayProperties.icon ?? SubmissionStatus.notSubmitted.icon }
+    var submissionColor: Color { .init(assignment.submission?.stateDisplayProperties.color ?? SubmissionStatus.notSubmitted.color) }
+    let defaultTextColor: Color = .textDark
+    let brandColor: Color = .init(Brand.shared.primary)
+    var hasPointsPossible: Bool { scoreLabel != nil }
+    var pointsPossibleText: String { assignment.pointsPossibleCompleteText }
 
-    public var icon: UIImage {
-        assignment.icon ?? .assignmentLine
+    // Teacher
+    var isTeacher: Bool { env.app == .teacher }
+    var needsGrading: Bool { assignment.needsGradingCount > 0 }
+    var needsGradingCount: Int { assignment.needsGradingCount }
+
+    private var formattedDueAt: String? {
+        guard let dueAt = assignment.dueAt else { return nil }
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "LLL d, y h:mm a"
+        return dateFormatter.string(from: dueAt)
     }
 
-    public var name: String {
-        assignment.name
+    var scoreLabel: String? {
+        guard let pointsPossible = assignment.pointsPossible else { return nil }
+        var scoreString = "-"
+        if let viewableScore = assignment.viewableScore {
+            scoreString = String(Int(viewableScore))
+        }
+        return "\(scoreString)/\(String(Int(pointsPossible)))"
     }
 
-    public var published: Bool? {
+    var published: Bool? {
```

Comment:
Please either make the class itself not public (I guess this won't be the case here), or ensure all it's non-private properties are consistently public.
So please keep them public, and also make the new ones public, unless private.